client/src/test/java/nl/altindag/client/ClientConfigShould.java

```diff
@@ -284,6 +292,59 @@ void createOldJerseyClientWithSecurity() {
         client.destroy();
     }
 
+    @Test
+    void createCxfJaxRsClientWithoutSecurity() {
+         javax.ws.rs.client.Client client = victim.cxfJaxRsClient(null);
+
+         assertThat(client).isNotNull();
+         assertThat(client.getClass().getPackageName())
+     		.as("CXF JAX-RS implemenatsion is used")
+     		.startsWith("org.apache.cxf");
+
+         assertThat(client.getSslContext()).isNull();
+         assertThat(client.getHostnameVerifier()).isNull();
+
+         client.close();
+    }
+
+    @Test
+    void createCxfJaxRsClientWithSecurity() {
+         SSLFactory sslFactory = createSSLFactory(false, true);
+
+         javax.ws.rs.client.Client client = victim.cxfJaxRsClient(sslFactory);
+
+         assertThat(client).isNotNull();
+         verify(sslFactory, times(1)).getSslContext();
+         verify(sslFactory, times(1)).getHostnameVerifier();
+
+         assertThat(client.getSslContext()).isNotNull();
+         assertThat(client.getHostnameVerifier()).isNotNull();
+
+
+         client.close();
+    }
+
+    @Test
+    void createCxfWebClientWithoutSecurity() {
+        org.apache.cxf.jaxrs.client.WebClient client = victim.cxfWebClient(null);
+
+         assertThat(client).isNotNull();
+
+         client.close();
+    }
+
+    @Test
+    void createCxfWebClientWithSecurity() {
+         SSLFactory sslFactory = createSSLFactory(false, true);
+
+         org.apache.cxf.jaxrs.client.WebClient client = victim.cxfWebClient(sslFactory);
+
+         assertThat(client).isNotNull();
+         // Noting sensible can be tested here since SSL factory is called when requests are made
```

Comment:
Than we can delete this unit test as this is a duplicate

Reply:
@Hakky54  the tests is model after all other tests setups  like createOldJerseyClientWithoutSecurity
Reasons to keep:
- can be keeped for consistency with other tests
- if fact it trigger different branch in config. :)    if (nonNull(sslFactory))

Reply:
Than let's keep it

Reply:
I did an attempt to verify if the ssl materials will be used provided by the configuration, see below for an example:
```java
@Test
void createCxfWebClientWithSecurity() {
    SSLFactory sslFactory = createSSLFactory(false, true);

    org.apache.cxf.jaxrs.client.WebClient client = victim.cxfWebClient(sslFactory);

    assertThat(client).isNotNull();

    assertThatThrownBy(() -> client.to(TestConstants.HTTPS_URL, false).get());

    verify(sslFactory, times(1)).getSslSocketFactory();
    verify(sslFactory, times(1)).getHostnameVerifier();

    client.close();
}
```
The client will try to execute a request with with a https url. This would force the client to use the ssl configuration. The request will fail because the server is not running, but this does not matter because we just only want to validate if the getSslSocketFactory and getHostnameVerifier methods will be called by the client configuration when sslfactory is present. The client will throw a runtime exception, but the assertThatThrownBy method will suppress that. Afterwords we can validate with the mockito verify methods if the getSslSocketFactory and getHostnameVerifier of the sslfactory has been used by the CXF client.